### PR TITLE
Upgrade to .NET 4.6.1

### PR DESCRIPTION
**Pull request by jayhilden**
- Update to the latest version of .NET (pre-CORE)
- Update to Visual Studio 2015
- Update build directory to the standard `/bin/release`

### CODE REVIEW

**Comment by niemyjski** [comment]
Is there any specific reason for changing this to 4.6.1? You can target 4.5 in a 4.6.1 project?

**Comment by jayhilden** [comment]
The main reason is that 4.5 has reached end of life.

https://blogs.msdn.microsoft.com/dotnet/2015/12/09/support-ending-for-the-net-framework-4-4-5-and-4-5-1/

So really, all projects should be updated to .NET 4.6.x and/or .NET CORE

**Comment by niemyjski** [comment]
We'd be more than happy to accept an updated pr, that adds a build target for 4.6 but we still have to have support for existing versions as not everyone updates :. We still are supporting users on VS2010.

File: Source/CodeSmith.Data/CodeSmith.Data.csproj [hunk 1]
```diff
@@ -9,30 +9,30 @@
...
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
     <TargetFrameworkProfile />
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>..\..\Common\$(TargetFrameworkVersion)\</OutputPath>
+    <OutputPath>bin\Debug\</OutputPath>
```
Comment (niemyjski) [inline Source/CodeSmith.Data/CodeSmith.Data.csproj:19]: Can you revert this change, we source import this git branch and do builds.

File: Source/CodeSmith.Data/CodeSmith.Data.csproj [hunk 2]
```diff
@@ -9,30 +9,30 @@
...
     <DefineConstants>TRACE;DEBUG;v40</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <DocumentationFile>..\..\Common\$(TargetFrameworkVersion)\CodeSmith.Data.XML</DocumentationFile>
+    <DocumentationFile>bin\Debug\CodeSmith.Data.XML</DocumentationFile>
```
Comment (niemyjski) [inline Source/CodeSmith.Data/CodeSmith.Data.csproj:23]: Can you revert this change, we source import this git branch and do builds.

File: Source/CodeSmith.Data/CodeSmith.Data.csproj [hunk 3]
```diff
@@ -9,30 +9,30 @@
...
     <NoWarn>1591</NoWarn>
     <CodeAnalysisRuleSet>AllRules.ruleset</CodeAnalysisRuleSet>
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>..\..\Common\$(TargetFrameworkVersion)\</OutputPath>
+    <OutputPath>bin\Release\</OutputPath>
```
Comment (niemyjski) [inline Source/CodeSmith.Data/CodeSmith.Data.csproj:31]: Can you revert this change, we source import this git branch and do builds.

File: Source/CodeSmith.Data/CodeSmith.Data.csproj [hunk 4]
```diff
@@ -9,30 +9,30 @@
...
     <DefineConstants>TRACE;v40</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <DocumentationFile>..\..\Common\$(TargetFrameworkVersion)\CodeSmith.Data.XML</DocumentationFile>
+    <DocumentationFile>bin\Release\CodeSmith.Data.XML</DocumentationFile>
```
Comment (niemyjski) [inline Source/CodeSmith.Data/CodeSmith.Data.csproj:35]: Can you revert this change, we source import this git branch and do builds.

File: Source/CodeSmith.Data/CodeSmith.Data.csproj [hunk 5]
```diff
@@ -67,9 +67,6 @@
     </Reference>
   </ItemGroup>
   <ItemGroup>
```
Comment (niemyjski) [inline Source/CodeSmith.Data/CodeSmith.Data.csproj:69]: Can you revert this change, we source import this git branch and do builds.

**Review by niemyjski — CHANGES_REQUESTED** [review]

**Comment by niemyjski** [comment]
At this time, we have no intention of merging this pull request with out the requested changes being made. Also, the source for CodeSmith.Data.Linq should be open source. Please feel free to reopen the pull request with the suggested changes

**Pull request closed without merging**